test/e2e/prometheusagent_test.go

```diff
@@ -365,3 +374,249 @@ func testPromAgentReconcileDaemonSetResourceDelete(t *testing.T) {
+	prometheusAgentDS.Spec.ScrapeInterval = "1s"
+
+	_, err = framework.CreatePrometheusAgentAndWaitUntilReady(ctx, ns, prometheusAgentDS)
+	require.NoError(t, err)
+
+	var pollErr error
+	err = wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(_ context.Context) (bool, error) {
+		ctx := context.Background()
+
+		appPods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{
+			LabelSelector: "group=test",
+		})
+		if err != nil {
+			pollErr = fmt.Errorf("can't list app pods: %w", err)
+			return false, nil
+		}
+
+		appPodsNodes := make([]string, 0, 2)
+		appPodsIPs := make([]string, 0, 2)
+		for _, pod := range appPods.Items {
+			appPodsNodes = append(appPodsNodes, pod.Spec.NodeName)
+			appPodsIPs = append(appPodsIPs, pod.Status.PodIP)
+		}
+
+		paPods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+		if err != nil {
+			pollErr = fmt.Errorf("can't list prometheus agent pods: %w", err)
+			return false, nil
+		}
+
+		paPodsNodes := make([]string, 0, 2)
+		for _, pod := range paPods.Items {
+			paPodsNodes = append(paPodsNodes, pod.Spec.NodeName)
+		}
+
+		if len(appPodsNodes) != len(paPodsNodes) {
+			pollErr = fmt.Errorf("got %d application pods and %d prometheus-agent pods", len(appPodsNodes), len(paPodsNodes))
+			return false, nil
+		}
+		for _, n := range appPodsNodes {
+			if !slices.Contains(paPodsNodes, n) {
+				pollErr = fmt.Errorf("no prometheus-agent pod found on node %s", n)
+				return false, nil
+			}
+		}
+
+		cfg := framework.RestConfig
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+
+		closer, err := testFramework.StartPortForward(ctx, cfg, "https", paPods.Items[0].Name, ns, "9090")
+		if err != nil {
+			pollErr = fmt.Errorf("can't start port forward to first prometheus agent pod: %w", err)
+			return false, nil
+		}
+		defer closer()
+
+		req, err := http.NewRequestWithContext(ctx, "GET", "https://localhost:9090/api/v1/targets", nil)
+		if err != nil {
+			pollErr = fmt.Errorf("can't create http request to first prometheus server: %w", err)
+			return false, nil
+		}
+
+		// The prometheus certificate is issued to <pod>.<namespace>.svc,
+		// but port-forwarding is done through localhost.
+		// This is why we use an http client which skips the TLS verification.
+		// In the test we will verify the TLS certificate manually to make sure
+		// the prometheus instance is configured properly.
+		transport := &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+		err = http2.ConfigureTransport(transport)
+		if err != nil {
+			pollErr = fmt.Errorf("can't configure transport: %w", err)
+			return false, nil
+		}
+
+		httpClient := http.Client{
+			Transport: transport,
+		}
+
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			pollErr = fmt.Errorf("can't send http request to first prometheus server: %w", err)
```

Comment:
@ArthurSens @simonpasquier The test is now failing here: `can't send http request to first prometheus server: Get "https://localhost:9090/api/v1/targets": http: server gave HTTP response to HTTPS client`

So AFAIU to fix this there are two options:

- Make http client receive http response
- Make `https://localhost:9090/api/v1/targets` serve https response

Seems like option 1 is the way to go right. If we follow this option I need to remove custom transport as stated in https://github.com/prometheus-operator/prometheus-operator/pull/6879#discussion_r1742064354 right?

Reply:
see my comment above (we need to fix the url scheme).